content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,320 @@
+ If `protocol-id` is not equal to `0`, `10`, `11`, `12`, `13`, this field MUST be set to `0`;
+ - `handshake-message`: the Noise handshake message (`handshake-message-len` bytes). 
+If `handshake-message-len` is not `0`, 
+it contains the concatenation of one or more Noise Diffie-Hellman ephemeral or static keys 
+encoded as in [Public Keys Encoding](#Public-Keys-Encoding);
+ - `transport-message-len-len`: the length in bytes of `transport-message-len` (1 byte);
+ - `transport-message-len`: the length in bytes of `transport-message` (`transport-message-len-len` bytes);
+ - `transport-message`: the transport message (`transport-message-len` bytes);
+ - `transport-message-ad`: the symmetric encryption authentication data for `transport-message` (16 bytes).
+
+
+### ABNF
+
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```abnf
+; protocol ID
+protocol-id                 = 1OCTET
+
+; contains the size of handshake-message
+handshake-message-len       = 1OCTET
+
+; containes one or more Diffie-Hellman public keys
+handshake-message           = *OCTET
+
+; contains the size of message-len
+transport-message-len-len   = 1OCTET
+
+; contains the size of transport-message
+transport-message-len       = *OCTET
+
+; contains the transport message, eventually encrypted
+transport-message           = *OCTET
+
+; contains authentication data for transport-message, if encrypted
+transport-message-ad        = 16OCTET
+
+; the Waku WakuMessage payload field
+payload  = protocol-id handshake-message-len handshake-message transport-message-len-len transport-message-len transport-message transport-message-ad 
+```
+
+### Protocol Payload Format
+
+Based on the specified `protocol-id`, 
+the Waku message `payload` field will encode different types of protocol-dependent messages. 
+
+In particular, if `protocol-id` is
+- `0`:  payload encodes an [after-handshake](#After-handshake) message.  
+    - `handshake-message-len` MAY be 0;
+    -  `transport-message` contains the Noise transport message;
+- `10`,`11`,`12`,`13`: payload encodes a supported Noise handshake message.  
+    -  `transport-message` contains the Noise transport message;
+- `30`: payload encapsulate a `ChaChaPoly` ciphertext `ct`.  
+    - `handshake-message-len` is set to `0`;
+    - `transport-message` contains the concatenation of the encryption nonce (12 bytes) followed by the ciphertext `ct`;
+    - `transport-message-len-len` and `transport-message-len` are set accordingly to `transport-message` length;
+    - `transport-message-ad` contains the authentication data for `ct`.
+
+
+### Public Keys Serialization
+
+Diffie-Hellman public keys can be trasmitted in clear 
+or in encrypted form with authentication data attached. 
+To distinguish between these two cases, public keys are serialized as the concatenation of the following three fields:
+
+- `flag`: 
```

Comment:
I see you addressed this comment, thanks, though it would be easier to see if the comment is addressed if you please reply to the comment (just to indicate that the comment is seen), and it would be greater if you include the commit id in which you have addressed the comment

Reply:
Sure! This was addressed in https://github.com/vacp2p/rfc/pull/496/commits/b587e8b2cb05ce1fa6042c2417696cb50be6fcf4